Authenticator/Source/SegmentedControlRowModel.swift

```diff
@@ -25,63 +25,46 @@
 
 import OneTimePassword
 
-protocol SegmentedControlRowModel {
-    typealias Action
+struct SegmentedControlRowModel<Action> {
+    let segments: [(title: String, action: Action)]
+    let selectedSegmentIndex: Int?
 
-    var segments: [(title: String, action: Action)] { get }
-    var selectedSegmentIndex: Int? { get }
+    init<V: Equatable>(options: [(title: String, value: V)], value: V, @noescape changeAction: (V) -> Action) {
+        segments = options.map({ option in
+            (title: option.title, action: changeAction(option.value))
+        })
+        selectedSegmentIndex = options.map({ $0.value }).indexOf(value)
+    }
 }
 
 enum TokenType {
     case Counter, Timer
 }
 
-struct TokenTypeRowModel<Action>: SegmentedControlRowModel {
-    let segments: [(title: String, action: Action)]
-    let selectedSegmentIndex: Int?
-
-    init(value: TokenType, @noescape changeAction: (TokenType) -> Action) {
+extension SegmentedControlRowModel {
+    init(tokenType value: TokenType, @noescape changeAction: (TokenType) -> Action) {
         let options = [
             (title: "Time Based", value: TokenType.Timer),
             (title: "Counter Based", value: TokenType.Counter),
         ]
-        segments = options.map({ option in
-            (title: option.title, action: changeAction(option.value))
-        })
-        selectedSegmentIndex = options.map({ $0.value }).indexOf(value)
+        self.init(options: options, value: value, changeAction: changeAction)
     }
-}
 
-struct DigitCountRowModel<Action>: SegmentedControlRowModel {
-    let segments: [(title: String, action: Action)]
-    let selectedSegmentIndex: Int?
-
-    init(value: Int, @noescape changeAction: (Int) -> Action) {
+    init(digitCount value: Int, @noescape changeAction: (Int) -> Action) {
         let options = [
             (title: "6 Digits", value: 6),
             (title: "7 Digits", value: 7),
             (title: "8 Digits", value: 8),
         ]
-        segments = options.map({ option in
-            (title: option.title, action: changeAction(option.value))
-        })
-        selectedSegmentIndex = options.map({ $0.value }).indexOf(value)
+        self.init(options: options, value: value, changeAction: changeAction)
     }
-}
 
-struct AlgorithmRowModel<Action>: SegmentedControlRowModel {
-    let segments: [(title: String, action: Action)]
-    let selectedSegmentIndex: Int?
-
-    init(value: Generator.Algorithm, @noescape changeAction: (Generator.Algorithm) -> Action) {
+    init(algorithm value: Generator.Algorithm, @noescape changeAction: (Generator.Algorithm) -> Action) {
```

Comment:
Line Length Violation: Line should be 100 characters or less: currently 105 characters (line_length)